java/core/src/main/java/co/worklytics/psoxy/storage/impl/ColumnarBulkDataSanitizerImpl.java

```diff
@@ -151,37 +155,51 @@ public byte[] sanitize(@NonNull InputStreamReader reader,
 
 
         try(ByteArrayOutputStream baos = new ByteArrayOutputStream(1024);
-            PrintWriter printWriter = new PrintWriter(baos);
+            Writer printWriter = new PrintWriter(baos);
             CSVPrinter printer = new CSVPrinter(printWriter, CSVFormat.DEFAULT
                 .withHeader(columnNamesForOutputFile.toArray(new String[0])))
             ) {
 
-            records.forEach(row -> {
-                Stream<Object> sanitized =
+            UnmodifiableIterator<List<CSVRecord>> chunks =
+                Iterators.partition(records.iterator(), rules.getRecordShuffleChunkSize());
+
```

Comment:
this previous step of splitting the rows into sublists is not really needed... in the end the whole process is held in memory (output = baos.toByteArray()) so could just do the shuffle on the whole file I think

Reply:
yeah, but I want to change that because it's stupid. So I don't want to *depend* on that behavior, as it blocks us from processing large files. (already broke asana import or something until they inc'd mem size in the lambda)

Reply:
could just read line by line and write in buffers of random size.
// new list size of random(50, 500)
// do { process and add to the list } while (list is not full && more to process)
// shuffle and write

Reply:
yeah, i don't know that random buffer size really helps anonymity in theory; but I guess it does in practice, just because if I was trying to attack this, it's one more issue I'd have to consider when trying to re-identify the data (eg, that the corpus of potential matches gets larger the deeper I go into the data set).  but I think randomly shuffle sets of 500 is enough for me to attack something *other* than order first when trying to re-identify.